addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputWithPlayControlXML.java

```diff
@@ -0,0 +1,223 @@
+        if (doc.getFirstChild() == null) {
+            throw new ReceivedMessageParseException("<Play_Info>GetParam failed: " + response);
+        }
+
+        PlayInfoState msg = new PlayInfoState();
+
+        Node playInfoNode = XMLUtils.getNode(doc.getFirstChild(), "Play_Info");
+
+        msg.playbackMode = XMLUtils.getNodeContentOrDefault(playInfoNode, "Playback_Info", "");
+
+        Node metaInfoNode = XMLUtils.getNode(playInfoNode, "Meta_Info");
+        if (metaInfoNode != null) {
+            String stationElement = inputID.equals(YamahaReceiverBindingConstants.INPUT_TUNER) ? "Radio_Text_A" : "Station";
+            msg.station = XMLUtils.getNodeContentOrDefault(metaInfoNode, stationElement, "");
+
+            msg.artist = XMLUtils.getNodeContentOrDefault(metaInfoNode, "Artist", "");
+            msg.album = XMLUtils.getNodeContentOrDefault(metaInfoNode, "Album", "");
+
+            String songElement = inputID.equals(YamahaReceiverBindingConstants.INPUT_SPOTIFY) ? "Track" : "Song";
+            msg.song = XMLUtils.getNodeContentOrDefault(metaInfoNode, songElement, "");
+        }
+
+        if (inputID.equals(YamahaReceiverBindingConstants.INPUT_SPOTIFY)) {
+            // Spotify input supports song cover image
+            String songImageUrl = XMLUtils.getNodeContentOrDefault(playInfoNode, "Album_ART/URL", "");
+            if (!StringUtils.isBlank(songImageUrl)) {
+                msg.songImageUrl = String.format("http://%s%s", com.getHost(), songImageUrl);
+            }
+        }
+
+        logger.trace("Playback: {}, Station: {}, Artist: {}, Album: {}, Song: {}, SongImageUrl: {}",
+                msg.playbackMode, msg.station, msg.artist, msg.album, msg.song, msg.songImageUrl);
+
+        observer.playInfoUpdated(msg);
+    }
+
+    /**
+     * Start the playback of the content which is usually selected by the means of the Navigation control class or
+     * which has been stopped by stop().
+     *
+     * @throws Exception
+     */
+    @Override
+    public void play() throws IOException, ReceivedMessageParseException {
+        sendPlaybackCommand("Play");
+    }
+
+    /**
+     * Stop the currently playing content. Use start() to start again.
+     *
+     * @throws Exception
+     */
+    @Override
+    public void stop() throws IOException, ReceivedMessageParseException {
+        sendPlaybackCommand("Stop");
+    }
+
+    /**
+     * Pause the currently playing content. This is not available for streaming content like on NET_RADIO.
+     *
+     * @throws Exception
+     */
+    @Override
+    public void pause() throws IOException, ReceivedMessageParseException {
+        sendPlaybackCommand("Pause");
+    }
+
+    /**
+     * Skip forward. This is not available for streaming content like on NET_RADIO.
+     *
+     * @throws Exception
+     */
+    @Override
+    public void skipFF() throws IOException, ReceivedMessageParseException {
+        if (inputID.equals(YamahaReceiverBindingConstants.INPUT_SPOTIFY)) {
```

Comment:
Position literals first in String comparisons - that way if the String is null you won't get a NullPointerException, it'll just return false. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Could you check all?